src/components/App/App.js

```diff
@@ -43,70 +81,175 @@ function App(props) {
         setNavMobileOpen(true);
     }
 
-    function handleSaveCard() {
-        if (setCardSaved(true)) {
-            setCardSaved(false);
-        } setCardSaved(true)
-    }
-
     function closeAllPopups() {
         setSuccessPopupOpen(false);
         setRegisterOpen(false);
         setAuthOpen(false);
+        history.push('/');
+    }
+
+    function closeNavMobile() {
         setNavMobileOpen(false);
     }
 
-    function handleRegister(e) {
-        e.preventDefault();
-        setSuccessPopupOpen(true);
-        setRegisterOpen(false);
+    // user forms submit
+    function handleAuth(email, password) {
+        mainApi.authorizeUser(email, password)
+            .then((res) => {
+                closeAllPopups();
+                setLoggedIn(true);
+                setCurrentUser({ username: res.user })
+                history.push('/saved-news');
+            })
+            .catch((e) => { console.log(e) });
+    }
+
+    function handleRegister(email, password, username) {
+        mainApi.registerUser(email, password, username)
+            .then(() => {
+                setRegisterOpen(false);
+                setSuccessPopupOpen(true);
+            })
+            .catch((e) =>
+                console.log(e));
+    }
+
+    // searching cards
+    function searchCards(search) {
+        setIsPreloaderOpen(true);
+        setIsNothingFoundOpen(false);
+        setCards([]);
+
+        newsApi.getCardsList(search)
+            .then((res) => {
+                if (res.totalResults === 0) {
+                    setIsNothingFoundOpen(true);
+                } else {
+                    setKeyword(search);
+                    setCards(res);
+                }
+            })
+            .then(() => {
+                setIsPreloaderOpen(false)
+            })
+            .catch((e) => console.log(e));
     }
 
-    function handleAuth(e) {
-        e.preventDefault();
-        closeAllPopups();
-        history.push('/saved-news')
+    // saving an article 
+    function handleSaveArticle(card) {
+        mainApi.saveArticle(card)
+            .then((res) => {
+                savedCards.push(res);
```

Comment:
`  savedCards.push(res);` 

you shouldn't mutate the array like this. Only the `state function` can do that. This function is `setSavedCards`.


```js

 setSavedCards(res.data);


```

You don't need to push anything